packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -335,7 +335,7 @@ class RawGestureDetectorState extends State<RawGestureDetector> {
   /// the gesture detector should be enabled.
   void replaceGestureRecognizers(Map<Type, GestureRecognizerFactory> gestures) {
     assert(() {
-      if (!RenderObject.debugDoingLayout) {
+      if (!Renderer.instance.pipelineOwner.debugDoingLayout) {
```

Comment:
We can be smarter about this once elements have owners, I think. I'll add a comment.